src/analyses/variable-sensitivity/variable_sensitivity_domain.cpp

```diff
@@ -0,0 +1,281 @@
+/*******************************************************************\
+
+Module: Abstract Interpretation
+
+Author: Martin Brain
+
+Date: April 2016
+
+\*******************************************************************/
+
+
+#include <util/message.h>
+
+#include "variable_sensitivity_domain.h"
+
+#ifdef DEBUG
+#include <iostream>
+#endif
+
+
+/*******************************************************************\
+
+Function: variable_sensitivity_domaint::transform
+
+  Inputs: The instruction before (from) and after (to) the abstract domain,
+          the abstract interpreter (ai) and the namespace (ns).
+
+ Outputs: None
+
+ Purpose: Compute the abstract transformer for a single instruction
+
+\*******************************************************************/
+
+void variable_sensitivity_domaint::transform(
+    locationt from,
+    locationt to,
+    ai_baset &ai,
+    const namespacet &ns)
+{
+  #ifdef DEBUG
+  std::cout << "Transform from/to:\n";
+  std::cout << from->location_number << " --> "
+            << to->location_number << std::endl;
+  #endif
+
+  const goto_programt::instructiont &instruction=*from;
+  switch(instruction.type)
+  {
+  case DECL:
+    // Creates a new variable, which should be top
+    // but we don't store top so ... no action required
+    break;
+
+  case DEAD:
+    {
+    // Assign to top is the same as removing
+    abstract_object_pointert top_object=
+      abstract_state.abstract_object_factory(
+        to_code_dead(instruction.code).symbol().type(), true);
+    abstract_state.assign(to_code_dead(instruction.code).symbol(), top_object);
+    }
+    break;
+
+  case ASSIGN:
+    {
+      const code_assignt &inst = to_code_assign(instruction.code);
+
+      // TODO : check return values
+      abstract_object_pointert r = abstract_state.eval(inst.rhs(), ns);
+      abstract_state.assign(inst.lhs(), r);
+    }
+    break;
+
+  case GOTO:
+    {
+      // TODO(tkiley): add support for flow sensitivity
+#if 0
+      if (flow_sensitivity == FLOW_SENSITIVE)
+      {
+        locationt next=from;
+        next++;
+        if(next==to)
+          assume(not_exprt(instruction.guard));
+        else
+          assume(instruction.guard);
+      }
+#endif
+    }
+    break;
+
+  case ASSUME:
+    abstract_state.assume(instruction.guard, ns);
+    break;
+
+  case FUNCTION_CALL:
+    // FIXME : Ignore as not yet interprocedural
+    break;
+
+  case END_FUNCTION:
+    // FIXME : Ignore as not yet interprocedural
+    break;
+
+    /***************************************************************/
+
+  case ASSERT:
+    // Conditions on the program, do not alter the data or information
+    // flow and thus can be ignored.
+    break;
+
+  case SKIP:
+  case LOCATION:
+    // Can ignore
+    break;
+
+  case RETURN:
+    throw "return instructions are depreciated";
```

Comment:
Nit: typo "depreciated" -> "deprecated" (I'm not actually sure they are, though)